src/components/EventCardsGridContainer/index.js

```diff
@@ -1,136 +0,0 @@
-/* eslint react/no-array-index-key: 0, no-nested-ternary: 0 */ // Disable "Do not use Array index in keys" for options since they dont have unique identifier
-
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
-import { withStyles } from '@material-ui/core';
-
-import styles from './styles';
-import { Token, OracleStatus, SortBy, EventStatus } from '../../constants';
-import graphqlActions from '../../redux/Graphql/actions';
-import EventCard from '../EventCard';
-import EventsEmptyBg from '../EventsEmptyBg';
-import InfiniteScroll from '../InfiniteScroll';
-
-const LIMIT = 24;
-const SKIP = 0;
-
-
-@withStyles(styles, { withTheme: true })
-@connect((state) => ({
-  topics: state.Graphql.get('getTopicsReturn'),
-  oracles: state.Graphql.get('getOraclesReturn'),
-  sortBy: state.Dashboard.get('sortBy'),
-  syncBlockNum: state.App.get('syncBlockNum'),
-  walletAddresses: state.App.get('walletAddresses'),
-  txReturn: state.Graphql.get('txReturn'),
-}), (dispatch) => ({
-  getOracles: (filters, orderBy, limit, skip, exclude) => dispatch(graphqlActions.getOracles(filters, orderBy, limit, skip, exclude)),
-}))
-export default class EventCardsGrid extends Component {
-  static propTypes = {
-    theme: PropTypes.object.isRequired,
-    topics: PropTypes.array,
-    getOracles: PropTypes.func,
-    oracles: PropTypes.array,
-    eventStatusIndex: PropTypes.number.isRequired,
-    sortBy: PropTypes.string,
-    syncBlockNum: PropTypes.number,
-    walletAddresses: PropTypes.array.isRequired,
-    status: PropTypes.string,
-    txReturn: PropTypes.object,
-  };
-
-  static defaultProps = {
-    txReturn: undefined,
-    topics: [],
-    getOracles: undefined,
-    oracles: [],
-    sortBy: SortBy.Ascending,
-    syncBlockNum: undefined,
-    status: '',
-  };
-
-  state = {
-    skip: 0,
-  }
-
-  componentWillMount() {
-    const { eventStatusIndex, sortBy, walletAddresses } = this.props;
-
-    this.executeGraphRequest(eventStatusIndex, sortBy, LIMIT, SKIP, walletAddresses);
-  }
-
-  componentWillReceiveProps(nextProps) {
-    const { eventStatusIndex, sortBy, syncBlockNum, walletAddresses, txReturn } = nextProps;
-
-    if (eventStatusIndex !== this.props.eventStatusIndex
-      || sortBy !== this.props.sortBy
-      || syncBlockNum !== this.props.syncBlockNum
-      || walletAddresses !== this.props.walletAddresses
-      || txReturn !== this.props.txReturn) {
```

Comment:
you are missing this logic in your BotCourt component:
```
|| syncBlockNum !== this.props.syncBlockNum
|| walletAddresses !== this.props.walletAddresses
|| txReturn !== this.props.txReturn) {
```

These are all important and they cause the page to update on:
1. a new block
2. when the wallet addresses change
3. when a tx is executed.

Reply:
wow, hm actually all 4 components who used `EventCardsGridContainer` are all missing now.... jesus. I will discuss this with alex about how to setup the MobX store structure for these things...

Reply:
Hmmm so its missing all this refreshing logic then... So for the time being, you will have to use the same logic with `componentWillReceiveProps` to make this work with the ones that aren't converted to MobX.

We already have `addresses` in the `WalletStore` so you will have to make a reaction in `BotCourtStore`:
```
reaction(
  () => this.app.wallet.addresses
  () => this.init() // refresh all the items
);
```

I have a PR for the `GlobalStore` that has the `syncBlockNum` so you can add a reaction for that as well when the PR gets merged.

And `txReturn` isnt converted to MobX yet so you will have to use the same logic as the deleted logic for now until that gets converted.

Reply:
oh, that's awesome, ok I will update them all and add reaction to every relevant components

Reply:
the whole idea of `txReturn` as a global variable doesn't make sense, discussed with alex and henry this afternoon. we decide to have a local variable like transactionFinished? to trigger an init.

Reply:
sounds good

Reply:
`QtumPredicition` is the only place that should have the `txReturn` mobx reaction. This is when the user creates a new event, the page should refresh based on the `txReturn` of the CreateEvent call.

Reply:
use freshing variable for it.